SUPPORTING_NEW_SERVICES.md

```diff
@@ -0,0 +1,108 @@
+* Be aware of differences in "update" behavior and name update/replace methods accordingly in your API.  See #321 for context.
+* Do not expose third party libraries in the API.  This has been a design choice from the beginning of the project, and all existing `gcloud-java` services adhere to this convention.
+* Member variable getters and builder setters should not use the prefixes "get" or "set".
+* Any service-generated IDs for model objects should be named `generatedId()`.
+
+#### SPI layer
+
+The SPI layer classes should be located in the package `com.google.cloud.servicename.spi`.  In most cases, the SPI layer should contain at least three classes:
+* An RPC factory interface (allows for the implementation to be loaded via the `java.util.ServiceLoader`).
+* An RPC interface that contains all RPC methods.
+* A default RPC implementation.
+
+#### Test helpers
+
+Test helper classes should be located in the package `com.google.cloud.servicename.testing`.  The naming convention for test helpers is `[Local|Remote][Service]Helper.java`.  For example, the local test helper for `gcloud-java-datastore` is named `LocalDatastoreHelper` and the remote test helper for `gcloud-java-storage` is named `RemoteStorageHelper`.  All test helpers should contain public `create` and `options` methods, and local helpers should contain `start` and `stop` methods.  See existing test helpers for information on what each of those methods should do.
+
+There are three types of test helpers:
+* When a local emulator is already available, your test helper should launch that emulator and return service options to connect to that local emulator.  This enables both users and our own library to run unit tests easily.  An example of this type of helper is `LocalDatastoreHelper`.  Google Cloud Datastore provides a script that launches a local datastore, so `LocalDatastoreHelper` launches that script in a separate process when the user calls `start()`.
+
+* When there is no local emulator but the service is simple enough to write an emulator, you should do so.  The emulator should listen to a port for requests, process those requests, and send responses back, being as true to the actual service as possible.  Dependencies in this mock should be as lightweight as possible.  Be sure to document differences between your emulator and the actual service.  Examples of this type of test helper are `LocalResourceManagerHelper` and `LocalDnsHelper`.
+
+* When there is no local emulator and the service is too complex to write a solid emulator, the test helper should contain methods to get options and separate test data from other user data.  `RemoteStorageHelper` is an example of this type of test helper, since there is no local emulator for Google Cloud Storage (at the time that this is written) and because the Google Cloud Storage API is complex.  `RemoteStorageHelper` has methods to:
```

Comment:
> at the time that this is written

Strange, for some reason "at the time this **was** written" sounds better to me however Google proves me wrong.

"at the time this is written" -> About 4,080,000 results
"at the time this was written" -> About 849,000 results